src/famfs_lib.c

```diff
@@ -236,24 +236,78 @@ famfs_print_uuid(const uuid_le *uuid)
 	printf("%s\n", uuid_str);
 }
 
-#define SYS_UUID_PATH "/sys/devices/virtual/dmi/id/product_uuid"
+#define SYS_UUID_FILE "/opt/famfs/system_uuid"
+#define SYS_UUID_DIR "/opt/famfs"
+
+/*
+ * Check if uuid file exists, if not, create it
+ * and update it with a new uuid.
+ *
+ */
+int famfs_create_sys_uuid_file()
+{
+	int uuid_fd, rc;
+	char uuid_str[37];  /* UUIDs are 36 char long, plus null terminator */
+	uuid_t local_uuid;
+	struct stat st = {0};
+	uuid_le sys_uuid;
+
+	/* Do nothing if file is present */
+	if (stat(SYS_UUID_FILE, &st) == 0)
+		return 0;
```

Comment:
If it's present, can we guarantee that it's valid?

Reply:
Have added a check to ensure it is a regular file. The content validation is done by the caller famfs_get_system_uuid().